### PR TITLE
Invert the dependency with hls-plugin-api

### PR DESCRIPTION
**Pull request by pepeiborra**
This PR includes changes to implement the reorg described in https://github.com/haskell/ghcide/issues/936#issuecomment-751437853

### CODE REVIEW

File: ghcide.cabal [hunk 1]
```diff
@@ -161,6 +162,9 @@ library
         Development.IDE.Plugin
         Development.IDE.Plugin.Completions
         Development.IDE.Plugin.CodeAction
+        Development.IDE.Plugin.Formatter
```
Comment (ndmitchell) [inline ghcide.cabal:165]: Why do these end up in Ghcide? I guess they leave in a future diff?

**Review by ndmitchell — APPROVED** [review]
I agree the repo inversion is the right thing to do. The details are less important and easier to clean up after, so stamping it with approval so you can do so. (I did review the patch though, seems sensible)

File: ghcide.cabal (hunk 1 above)
Comment (pepeiborra) [reply ghcide.cabal:165]: `Formatter` contains machinery needed to convert an HLS `FormattingProvider` into a `PartialHandler` for ghcide

**Comment by pepeiborra** [comment]
Closed in favour of https://github.com/haskell/haskell-language-server/pull/701/

**Pull request closed without merging**